x/pocketcore/keeper/service.go

```diff
@@ -12,8 +12,13 @@ import (
 // HandleRelay handles an api (read/write) request to a non-native (external) blockchain
```

Comment:
I ran this test (`go test -count=1 -v -run TestKeeper_HandleRelay ./x/pocketcore/keeper/...`) after making the following diff on your branch (`i.e. removing the logic for block height tolerance added in this PR`) which leads me to believe the change was not implemented correctly.

Am I missing something?

```diff
--- a/x/pocketcore/keeper/service.go
+++ b/x/pocketcore/keeper/service.go
@@ -3,21 +3,23 @@ package keeper
 import (
        "encoding/hex"
        "fmt"
+       "time"
+
        "github.com/pokt-network/pocket-core/crypto"
        sdk "github.com/pokt-network/pocket-core/types"
        pc "github.com/pokt-network/pocket-core/x/pocketcore/types"
-       "time"
 )

 // HandleRelay handles an api (read/write) request to a non-native (external) blockchain
 func (k Keeper) HandleRelay(ctx sdk.Ctx, relay pc.Relay) (*pc.RelayResponse, sdk.Error) {
        relayTimeStart := time.Now()

-       sessionBlockHeight := relay.Proof.SessionBlockHeight
+       // sessionBlockHeight := relay.Proof.SessionBlockHeight
+       sessionBlockHeight := k.GetLatestSessionBlockHeight(ctx)

-       if !k.IsLatestSessionHeightWithinTolerance(ctx, sessionBlockHeight) {
-               return nil, pc.NewInvalidBlockHeightError(pc.ModuleName)
-       }
+       // if !k.IsLatestSessionHeightWithinTolerance(ctx, sessionBlockHeight) {
+       //      return nil, pc.NewInvalidBlockHeightError(pc.ModuleName)
+       // }

        var node *pc.PocketNode
        // There is reference to node address so that way we don't have to recreate address twice for pre-leanpokt
diff --git a/x/pocketcore/keeper/session.go b/x/pocketcore/keeper/session.go
index 3225e55b..d0a5ab73 100644
--- a/x/pocketcore/keeper/session.go
+++ b/x/pocketcore/keeper/session.go
@@ -2,6 +2,7 @@ package keeper

 import (
        "encoding/hex"
+
        sdk "github.com/pokt-network/pocket-core/types"
        "github.com/pokt-network/pocket-core/x/nodes/exported"
        "github.com/pokt-network/pocket-core/x/pocketcore/types"
@@ -10,13 +11,9 @@ import (
 // "HandleDispatch" - Handles a client request for their session information
 func (k Keeper) HandleDispatch(ctx sdk.Ctx, header types.SessionHeader) (*types.DispatchResponse, sdk.Error) {
        // retrieve the latest session block height
-       var sessionBlockHeight int64
-       if types.GlobalPocketConfig.ClientSessionSyncAllowance > 0 && header.SessionBlockHeight != 0 {
-               sessionBlockHeight = header.SessionBlockHeight
-       } else {
-               sessionBlockHeight = k.GetLatestSessionBlockHeight(ctx)
-               header.SessionBlockHeight = sessionBlockHeight
-       }
+       latestSessionBlockHeight := k.GetLatestSessionBlockHeight(ctx)
+       // set the session block height
+       header.SessionBlockHeight = latestSessionBlockHeight

        // validate the header
        err := header.ValidateHeader()
@@ -24,7 +21,7 @@ func (k Keeper) HandleDispatch(ctx sdk.Ctx, header types.SessionHeader) (*types.
                return nil, err
        }
        // get the session context
-       sessionCtx, er := ctx.PrevCtx(sessionBlockHeight)
+       sessionCtx, er := ctx.PrevCtx(latestSessionBlockHeight)
        if er != nil {
                return nil, sdk.ErrInternal(er.Error())
        }
```

Reply:
Can you provide your test results?

Reply:
Ongoing discussion in main PR thread.